oak/server/storage/storage_channel.cc

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "storage_channel.h"
+
+namespace oak {
+
+StorageChannel::StorageChannel(std::unique_ptr<Storage::Stub> storage_service)
+    : storage_service_(std::move(storage_service)) {}
+
+uint32_t StorageChannel::Write(absl::Span<const char> request_data) {
+  grpc::Status status;
+  grpc::ClientContext context;
+  StorageOperationRequest operation_request;
+  StorageOperationResponse operation_response;
+
+  operation_request.ParseFromString(std::string(request_data.data(), request_data.size()));
+
+  switch (operation_request.operation_case()) {
+    case StorageOperationRequest::kReadRequest: {
+      ReadResponse response;
+      status = storage_service_->Read(&context, operation_request.read_request(), &response);
```

Comment:
Maybe you could even inline `operations_response.mutable_read_response()` here? Assuming that the method will not modify it unless the status was ok, which I think it's fair to assume (rather than having to repeat the check everywhere here).

Reply:
Good idea, thanks.